crates/e2e/tests/e2e/liquidity.rs

```diff
@@ -0,0 +1,409 @@
+        sell_amount: amount,
+        buy_token: token_usdt.address(),
+        buy_amount: amount - to_wei_with_exp(1, 8),
+        valid_to: model::time::now_in_epoch_seconds() + 300,
+        kind: OrderKind::Sell,
+        ..Default::default()
+    }
+    .sign(
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(trader.private_key()).unwrap()),
+    );
+
+    let zeroex_api_port = {
+        let order = order.clone();
+        let chain_id = web3.eth().chain_id().await.unwrap().as_u64();
+        let weth_addr = onchain.contracts().weth.address();
+        let gpv2_addr = onchain.contracts().gp_settlement.address();
+        let zeroex_addr = zeroex.address();
+        let orders_handler = Arc::new(Box::new(move |query: &OrdersQuery| {
+            orders_query_handler(
+                query,
+                order.clone(),
+                zeroex_maker.clone(),
+                zeroex_addr,
+                gpv2_addr,
+                chain_id,
+                weth_addr,
+            )
+        }));
+
+        ZeroExApi::builder()
+            .with_orders_handler(orders_handler)
+            .build()
+            .run()
+            .await
+    };
+
+    // Place Orders
+    let services = Services::new(onchain.contracts()).await;
+    let solver_endpoint =
+        colocation::start_baseline_solver(onchain.contracts().weth.address()).await;
+    colocation::start_driver_with_zeroex_liquidity(
+        onchain.contracts(),
+        vec![SolverEngine {
+            name: "test_solver".into(),
+            account: solver,
+            endpoint: solver_endpoint,
+        }],
+        zeroex_api_port,
+    );
+    services.start_autopilot(
+        None,
+        vec![
+            "--price-estimation-drivers=test_quoter|http://localhost:11088/test_solver".to_string(),
+            "--drivers=test_solver|http://localhost:11088/test_solver".to_string(),
+        ],
+    );
+    services
+        .start_api(vec![
+            "--price-estimation-drivers=test_quoter|http://localhost:11088/test_solver".to_string(),
+        ])
+        .await;
+    let order_id = services.create_order(&order).await.unwrap();
+    let limit_order = services.get_order(&order_id).await.unwrap();
+    assert_eq!(limit_order.metadata.class, OrderClass::Limit);
+
+    // Drive solution
+    tracing::info!("Waiting for trade.");
+    let sell_token_balance_before = token_usdc
+        .balance_of(trader.address())
+        .call()
+        .await
+        .unwrap();
+    let buy_token_balance_before = token_usdt
+        .balance_of(trader.address())
+        .call()
+        .await
+        .unwrap();
+
+    wait_for_condition(TIMEOUT, || async { services.solvable_orders().await == 1 })
+        .await
+        .unwrap();
+
+    wait_for_condition(TIMEOUT, || async { services.solvable_orders().await == 0 })
+        .await
+        .unwrap();
```

Comment:
Might be easier to just wait for buy token balance to no longer be 0

Reply:
Done